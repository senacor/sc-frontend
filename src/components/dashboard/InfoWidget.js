import { NavLink } from 'react-router-dom';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Card from '@material-ui/core/Card';
import Icon from '@material-ui/core/Icon';
import React from 'react';
import PropTypes from 'prop-types';

function InfoWidget(props) {
  const style = {
    margin: '20px',
    textDecoration: 'none',
    padding: '0',
    alignItems: 'stretch',
    height: '100%'
  };

  return (
    <Card
      component={props.linkTo ? NavLink : null}
      to={props.linkTo}
      style={style}
      onClick={props.onClick}
    >
      <CardContent
        style={{
          display: 'flex',
          flexDirection: 'row',
          padding: '0'
        }}
      >
        <div
          style={{
            flexGrow: '1',
            padding: '16px',
            fontSize: 30,
            alignSelf: 'center'
          }}
        >
          <Icon fontSize={'large'} color={'primary'}>
            {props.icon}
          </Icon>
        </div>
        <div
          style={{
            flexGrow: '3',
            padding: '16px 24px 16px 0px'
          }}
        >
          <Typography color="textSecondary">{props.label}</Typography>
          <Typography variant="h5" component="h2">
            {props.value}
          </Typography>
        </div>
      </CardContent>
    </Card>
  );
}

InfoWidget.propTypes = {
  onClick: PropTypes.func,
  value: PropTypes.node,
  label: PropTypes.node,
  icon: PropTypes.string,
  linkTo: PropTypes.string
};

export default InfoWidget;
