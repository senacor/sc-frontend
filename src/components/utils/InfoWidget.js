import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { withStyles } from '@material-ui/core';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Card from '@material-ui/core/Card';
import Icon from '@material-ui/core/Icon';

const styles = theme => ({
  card: {
    marginTop: 3 * theme.spacing.unit,
    marginLeft: 3 * theme.spacing.unit,
    textDecoration: 'none',
    height: 100,
    transition: '0.3s',
    '&:hover': {
      transform: 'scale(1.05)'
    }
  },
  cardContent: {
    display: 'flex',
    flexDirection: 'row',
    padding: 0
  },
  iconContainer: {
    flexGrow: 1,
    padding: 2 * theme.spacing.unit,
    fontSize: 30
  },
  textContainer: {
    width: 250,
    padding: 2 * theme.spacing.unit,
    paddingLeft: theme.spacing.unit
  }
});

const InfoWidget = ({ linkTo, onClick, icon, label, value, classes }) => {
  return (
    <Card
      component={linkTo && NavLink}
      to={linkTo}
      className={classes.card}
      onClick={onClick}
    >
      <CardContent className={classes.cardContent}>
        <div className={classes.iconContainer}>
          <Icon fontSize={'large'} color={'primary'}>
            {icon}
          </Icon>
        </div>
        <div className={classes.textContainer}>
          <Typography color="textSecondary">{label}</Typography>
          <Typography variant="h5" component="h2">
            {value}
          </Typography>
        </div>
      </CardContent>
    </Card>
  );
};

InfoWidget.propTypes = {
  classes: PropTypes.object.isRequired,
  onClick: PropTypes.func,
  value: PropTypes.node,
  label: PropTypes.node,
  icon: PropTypes.string,
  linkTo: PropTypes.string
};

export default withStyles(styles)(InfoWidget);
