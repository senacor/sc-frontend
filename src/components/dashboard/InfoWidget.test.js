import React from 'react';
import InfoWidget from './InfoWidget';
import { createShallow } from '@material-ui/core/test-utils';

describe('InfoWidget component', () => {
  let shallow = createShallow({ dive: false });

  it('should match snapshot', () => {
    let wrapper = shallow(
      <InfoWidget
        label={'PRs mit mir als Vorgesetzter insgesamt (inkl. Archiv)'}
        value={'3'}
        linkTo={'/prs'}
        onClick={() => {}}
        icon={'library_books'}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
