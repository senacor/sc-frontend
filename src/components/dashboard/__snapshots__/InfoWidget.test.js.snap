// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoWidget component should match snapshot 1`] = `
<WithStyles(Card)
  component={[Function]}
  onClick={[Function]}
  style={
    Object {
      "alignItems": "stretch",
      "height": "100%",
      "margin": "20px",
      "padding": "0",
      "textDecoration": "none",
    }
  }
  to="/prs"
>
  <WithStyles(CardContent)
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "padding": "0",
      }
    }
  >
    <div
      style={
        Object {
          "alignSelf": "center",
          "flexGrow": "1",
          "fontSize": 30,
          "padding": "16px",
        }
      }
    >
      <WithStyles(Icon)
        color="primary"
        fontSize="large"
      >
        library_books
      </WithStyles(Icon)>
    </div>
    <div
      style={
        Object {
          "flexGrow": "3",
          "padding": "16px 24px 16px 0px",
        }
      }
    >
      <WithStyles(Typography)
        color="textSecondary"
      >
        PRs mit mir als Vorgesetzter insgesamt (inkl. Archiv)
      </WithStyles(Typography)>
      <WithStyles(Typography)
        component="h2"
        variant="h5"
      >
        3
      </WithStyles(Typography)>
    </div>
  </WithStyles(CardContent)>
</WithStyles(Card)>
`;
